### PR TITLE
GEODE-5600: Unify SCM property caching.

### PR DESCRIPTION
**Pull request by PurelyApplied**
* Refactor gradle tasks generating .buildinfo
and GemFireVerion.properties files
* Treat .buildinfo as the source of truth for
those properties shared in GemFireVersion.properties

Co-authored-by: Patrick Rhomberg <prhomberg@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -101,6 +102,51 @@ task devBuild(dependsOn: [":assemble"]) {
   }
 }
 
+
```
Comment (metatype) [inline build.gradle]: Would it make sense to keep this in the utilities build script?

Comment (pivotal-jbarrett) [reply build.gradle]: Since it is only used in one place it didn’t make sense to me to have it located elsewhere.

**Comment by PurelyApplied** [comment]
@rhoughton-pivot @dickcav @smgoller to review?

Side note: I still don't understand why I can't tag some of you in the Review Requested section.

**Comment by pivotal-jbarrett** [comment]
@metatype ping!

**Comment by rhoughton-pivot** [comment]
I'd really prefer to not use `rootProject` in the root-project gradle file.

**Pull request merged**